features/step_definitions/machine.rb

```diff
@@ -1,10 +1,9 @@
 Given(/^I have an IPI deployment$/) do
   # In an IPI deployment, machine number equals to node number
   machines = BushSlicer::MachineMachineOpenshiftIo.list(user: admin, project: project("openshift-machine-api"))
-  if machines.length == 0
```

Comment:
We needs to:

1. handle the `Not an IPI deployment` situation
2. handle the `is an IPI but no worker machine` situation

This PR removes 1 and keeps 2, and the `logger.warn` is logging message irrelevant to the scenario `I have an IPI deployment`

Reply:
I added to make sure we exit even if any one of the above is not fulfilled PTAL.

Reply:
My point is what you are doing isn't what the scenario wants to do. The scenario is `Given I have an IPI deployment`, and what you are doing is skipping because the machine does not have a worker. Instead of removing the implementation of the scenario, you should add a condition to skip properly.

Also here `(machines.to_s.include?'worker')`: `machine` is an array but is transformed into a string, this is not a solid validation, at lest you need to iterate each machine object and decide by their label whether they are worker. Please refer to my earlier comment

> A more solid validation is checking you have a worker machineset, and check the replicas of the machineset isn't 0, then validate the number of workers equals the numbers of worker machines.